homeassistant/components/ebus/ebusd.py

```diff
@@ -0,0 +1,189 @@
+        command = READ_COMMAND.format(self._circuit, name, CACHE_TTL)
+
+        try:
+            _LOGGER.debug("Opening socket to ebusd %s: %s", name, command)
+            command_result = ebusdpy.send_command(self._address, command)
+            if 'not found' in command_result:
+                _LOGGER.warning("Element not found: %s", name)
+                raise RuntimeError("Element not found")
+            else:
+                self.value[name] = command_result
+        except socket.timeout:
+            _LOGGER.error("socket timeout error")
+            raise RuntimeError("socket timeout")
+        except socket.error:
+            _LOGGER.error("socket error: %s", socket.error)
+            raise RuntimeError("Command failed")
+
+    def write(self, call):
+        """Call write methon on ebusd."""
+        import ebusdpy
+        name = call.data.get('name')
+        value = call.data.get('value')
+        command = WRITE_COMMAND.format(self._circuit, name, value)
+
+        try:
+            _LOGGER.debug("Opening socket to ebusd %s: %s", name, command)
+            command_result = ebusdpy.send_command(self._address, command)
+            if 'done' not in command_result:
+                _LOGGER.warning('Write command failed: %s', name)
+        except socket.timeout:
+            _LOGGER.error("socket timeout error")
+        except socket.error:
+            _LOGGER.error()
+
+
+class Ebusd(Entity):
+    """Representation of a Sensor."""
+
+    def __init__(self, data, circuit, sensor_type, name):
+        """Initialize the sensor."""
+        self._state = None
+        self._client_name = name
+        self._name = SENSOR_TYPES[circuit][sensor_type][0]
+        self._unit_of_measurement = SENSOR_TYPES[circuit][sensor_type][1]
+        self._icon = SENSOR_TYPES[circuit][sensor_type][2]
+        self._type = SENSOR_TYPES[circuit][sensor_type][3]
+        self.data = data
+
+    @property
+    def name(self):
+        """Return the name of the sensor."""
+        return '{} {}'.format(self._client_name, self._name)
+
+    @property
+    def state(self):
+        """Return the state of the sensor."""
+        return self._state
+
+    @property
+    def icon(self):
+        """Icon to use in the frontend, if any."""
+        return self._icon
+
+    @property
+    def unit_of_measurement(self):
+        """Return the unit of measurement."""
+        return self._unit_of_measurement
+
+    def update(self):
+        """Fetch new state data for the sensor."""
+        try:
+            self.data.update(self._name)
+            if self._name not in self.data.value:
+                return
+
+            if self._type == 0:
+                self._state = format(
+                    float(self.data.value[self._name]), '.1f')
+            elif self._type == 1:
+                self._state = timer_format(self.data.value[self._name])
+            elif self._type == 2:
+                if self.data.value[self._name] == 1:
```

Comment:
I would also add the recognition of such a value as 'on' ` or self.data.value[self._name] == 'on' ` , , because frequent are just such answers

Reply:
there is the STATE_ON for that. In this way it will use also translation